### PR TITLE
sf3 compatibility

### PR DESCRIPTION
**Pull request by lnahiro**
use setFactory instead of setFactoryService and setFactoryMethod

### CODE REVIEW

**Comment by omansour** [comment]
fixes https://github.com/M6Web/AmqpBundle/issues/32

**Comment by Oliboy50** [comment]
@lnahiro you should change the composer.json to be sf3 compatible too
```
"symfony/framework-bundle" : "^2.7 || ^3.0",
"symfony/yaml": "^2.7.5 || ^3.0"
```

**Comment by twdmnd** [comment]
👍

File: .travis.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ matrix:
     - php: 5.5
       env: COMPOSER_FLAGS="--prefer-lowest"
     - php: 7.0
-      env: COMPOSER_FLAGS="--prefer-stable" SYMFONY_VERSION=2.7.*
+      env: COMPOSER_FLAGS="--prefer-stable" SYMFONY_VERSION=2.8.*
   fast_finish: true
```
Comment (Oliboy50) [inline .travis.yml]: We're currently triggering 5 builds Travis... but no one is built using SF3.

I propose to replace the whole travis.yml file with the following (which will trigger 6 builds and covers all our requirements IMHO):

```
language: php

php:
  - 5.5
  - 7.0

env:
  - SYMFONY_VERSION="2.8.*"
  - SYMFONY_VERSION="3.1.*"
  - COMPOSER_FLAGS="--prefer-lowest"

matrix:
  fast_finish: true

before_install:
  - phpenv config-rm xdebug.ini
  - bash travis/install_amqp.sh
  - composer self-update
  - if [ "$SYMFONY_VERSION" != "" ]; then composer require symfony/symfony:${SYMFONY_VERSION} --no-update; fi

install: composer update --prefer-dist $COMPOSER_FLAGS

script: bin/atoum
```

File: .travis.yml [hunk 2]
```diff
@@ -12,7 +12,7 @@ matrix:
     - php: 5.5
       env: COMPOSER_FLAGS="--prefer-lowest"
     - php: 7.0
-      env: COMPOSER_FLAGS="--prefer-stable" SYMFONY_VERSION=2.7.*
+      env: COMPOSER_FLAGS="--prefer-stable" SYMFONY_VERSION=2.8.*
```
Comment (Oliboy50) [inline .travis.yml]: We're currently triggering 5 builds Travis... but no one is built using SF3.

I propose to replace the whole travis.yml file with the following (which will trigger 6 builds and covers all our requirements IMHO):

```
language: php

php:
  - 5.5
  - 7.0

env:
  - SYMFONY_VERSION="2.8.*"
  - SYMFONY_VERSION="3.1.*"
  - COMPOSER_FLAGS="--prefer-lowest"

matrix:
  fast_finish: true

before_install:
  - phpenv config-rm xdebug.ini
  - bash travis/install_amqp.sh
  - composer self-update
  - if [ "$SYMFONY_VERSION" != "" ]; then composer require symfony/symfony:${SYMFONY_VERSION} --no-update; fi

install: composer update --prefer-dist $COMPOSER_FLAGS

script: bin/atoum
```

Comment (lnahiro) [inline .travis.yml]: done

**Comment by Oliboy50** [comment]
👍 great job :)

**Comment by fabdsp** [comment]
:+1:

**Pull request merged**